lib/publiccloud/gce.pm

```diff
@@ -81,8 +82,56 @@ sub upload_img {
     return $img_name;
 }
 
-sub cleanup {
-    record_info('info', 'to be implemented');
+sub ipa {
+    my ($self, %args) = @_;
+
+    $args{instance_type}        //= 'n1-standard-2';
+    $args{cleanup}              //= 1;
+    $args{ssh_private_key_file} //= '.ssh/id_rsa';
+    $args{tests}                //= '';
+    $args{timeout}              //= 60 * 20;
+    $args{results_dir}          //= 'ipa_results';
+
+    $args{tests} =~ s/,/ /g;
+
+    if (script_run('test -f ' . $args{ssh_private_key_file}) != 0) {
+        assert_script_run('SSH_DIR=`dirname ' . $args{ssh_private_key_file} . '`; test -d $SSH_DIR || mkdir -p $SSH_DIR');
```

Comment:
```suggestion
        assert_script_run('SSH_DIR=`dirname ' . $args{ssh_private_key_file} . '`; mkdir -p $SSH_DIR');
```

is better and doing exactly the same. have checked manually in case you call `mkdir` with `-p` even if directory exists it return 0